### PR TITLE
[Lineage] Fix artifact name empty bug

### PR DESCRIPTION
**Pull request by Bobgy**
Issue reported in https://github.com/kubeflow/pipelines/issues/3625

Root cause is that: we are using `instance of Artifact` to determine a resource if an artifact or an execution, but that doesn't work in transpiled javascript. So it was actually calling getExecutionName on an artifact.

Before, it worked because getting name from Artifact and Execution was the same, but I think after some iterations, TFX has moved name from a property to a custom property (or vice versa, I'm not sure).
After discussing with @Ark-kun, we decided to let UI always read properties from both custom property and property, so it will be more robust.

My solution for the typing problem is to use typescript native typing to let us know clearly what the resource type is and call corresponding methods.

Before the fix (note all artifacts are unnamed):
bug

After the fix (note artifacts have a proper name now): 
1DmcX14bB3N

### CODE REVIEW

File: src/mlmd/LineageTypes.ts
Comment (avdaredevil) [inline src/mlmd/LineageTypes.ts:7]: How are these different?

File: src/mlmd/Utils.tsx
Comment (avdaredevil) [inline src/mlmd/Utils.tsx]: Preserve comma
```suggestion
    ['RUN_ID', 'RUN', 'PIPELINE_NAME', 'WORKSPACE'],
```

**Review by avdaredevil — CHANGES_REQUESTED** [review]

**Comment by jlewi** [comment]
I'll let @avdaredevil review this since I don't know much about front ends.

File: src/mlmd/Utils.tsx
Comment (Bobgy) [reply src/mlmd/Utils.tsx]: SGTM, shall I set up a auto-formatter quickly?

File: src/mlmd/LineageTypes.ts
Comment (Bobgy) [reply src/mlmd/LineageTypes.ts:7]: Not really different, in an intermediate state, I updated this type to use type expressions, but in the end. It was fine to keep it as interface.

I'll restore unneeded change here.

**Review by Bobgy — COMMENTED** [review]
@avdaredevil updated, please review again

**Comment by Bobgy** [comment]
hmmm, suggestion feature still conflicts with google cla bot. I had to fix the commit

**Comment by Bobgy** [comment]
@googlebot I fixed it

File: src/mlmd/LineageView.tsx
Comment (avdaredevil) [inline src/mlmd/LineageView.tsx]: If we're using auto-styles could we keep it matched to other styles in the codebase, ie:
```suggestion
        typedResource: {type: 'artifact', resource: artifact},
```

File: src/mlmd/LineageCardRow.tsx
Comment (avdaredevil) [inline src/mlmd/LineageCardRow.tsx]: Same interface change here

File: src/mlmd/LineageTypes.ts
Comment (avdaredevil) [inline src/mlmd/LineageTypes.ts]: Same Bracket change
```suggestion
    | {type: 'artifact', resource: Artifact}
```

File: src/mlmd/Utils.tsx
Comment (avdaredevil) [inline src/mlmd/Utils.tsx]: Same

File: src/mlmd/LineageTypes.ts
Comment (avdaredevil) [inline src/mlmd/LineageTypes.ts]: Same

**Review by avdaredevil — CHANGES_REQUESTED** [review]
Hey @Bobgy, looks great! A few more nits, I'll approve in advance

/approve

**Comment by avdaredevil** [comment]
/lgtm

Holding for changes aforementioned
/hold

File: src/mlmd/LineageView.tsx
Comment (Bobgy) [reply src/mlmd/LineageView.tsx]: If we use prettier, it's not possible. Prettier is very opinionated about the format, it was invented to stop everyone from arguing about format and just use the common one (and bear with whatever you don't like about it, to be honest, I like their formatting very much, so it's not really a problem for me).
https://prettier.io/docs/en/options.html

I think that saves engineer resources from my previous experience and also when working on kubeflow/pipelines UI.

File: src/mlmd/LineageCardRow.tsx
Comment (Bobgy) [reply src/mlmd/LineageCardRow.tsx]: Done

File: src/mlmd/LineageView.tsx
Comment (Bobgy) [reply src/mlmd/LineageView.tsx]: what do you think? I'm following your suggestion for this PR

File: src/mlmd/LineageTypes.ts
Comment (Bobgy) [reply src/mlmd/LineageTypes.ts]: Done

File: src/mlmd/Utils.tsx
Comment (Bobgy) [reply src/mlmd/Utils.tsx]: Done

**Comment by Bobgy** [comment]
Thanks for the review!
@avdaredevil @jlewi Can you give an lgtm again? I fixed the formatting.
/unhold

File: src/mlmd/LineageView.tsx
Comment (avdaredevil) [reply src/mlmd/LineageView.tsx]: Yeah I agree with consistency, so if it's an automated tool, we can enforce those styles across the codebase. SGTM :)

**Review by avdaredevil — APPROVED** [review]

**Comment by avdaredevil** [comment]
/lgtm
/approve

Thanks @Bobgy!

**Pull request merged**
